### DISCUSSION TITLE
`The write operation timed out` when transcribing 3 hour .mp4 file

### DISCUSSION THREAD

**Discussion by KentonParton** [deepgram/community#521] [General help] [answered]
Hi All,

I am trying to transcribe a local `.mp4` file but I am encountering the error `The write operation timed out`. This is the code:

```python
from deepgram import DeepgramClient, PrerecordedOptions, FileSource

DEEPGRAM_API_KEY = ''

AUDIO_FILE = ''
UNLOAD_PATH = './transcriptions/'


def main():
    try:

        with open(AUDIO_FILE, "rb") as file:
            buffer_data = file.read()

        payload: FileSource = {
            "buffer": buffer_data,
        }
        deepgram = DeepgramClient(DEEPGRAM_API_KEY)

        options = PrerecordedOptions(
            model="nova-2",
            language="en",
            smart_format=True,
            punctuate=True,
            paragraphs=True,
            diarize=True,
            detect_topics=True,
        )

        video_id = 'some-id'

        if video_id is None:
            raise Exception('Could not extract video id')

        file_response = deepgram.listen.prerecorded.v("1").transcribe_file(payload, options)
        print(file_response)

        response_dict = file_response.to_dict()


        with open(f'{UNLOAD_PATH}/{video_id}.json', 'w') as file:
            json.dump(response_dict, file, indent=4)

    except Exception as e:
        print(f'Exception: {e}')


if __name__ == '__main__':
    main()

```
I also tried uploading the same file in the Deepgram console and ran into an error.

Does Deepgram have an audio length limit and is there a way around it? Or is it a code issue on my end? Thanks!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @KentonParton 

A 3 hour (what looks like a video) file is very very large and will exceed the defaults of what individuals typically use. You need to increase the timeout on the operation which you can do here:
https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/prerecorded/v1/client.py#L162

Also, if you want to speed that up (for your own sanity), you might want to strip out the video and only send the audio. It's been a while doing it by hand, but I think this would work `ffmpeg -i video.mp4 -vn audio.mp3`.
